squirrel.worker/src/main/java/org/dice_research/squirrel/worker/impl/WorkerImpl.java

```diff
@@ -328,15 +330,17 @@ public void sendNewUris(Iterator<byte[]> uriIterator) {
                 uriProcessor.recognizeUriType(newUri);
                 newUris.add(newUri);
                 if ((newUris.size() >= (packageCount + 1) * MAX_URIS_PER_MESSAGE) && uriIterator.hasNext()) {
+                    List<CrawleableUri> distinctUris = newUris.stream().distinct().collect(Collectors.toList());
```

Comment:
This change is not necessary. It is not the task of the worker to ensure that the URIs are distinct. In fact, it will assume that the URI collector did this already. So please undo this change.